trace_replay/trace_record_result.cc

```diff
@@ -0,0 +1,98 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#include "rocksdb/trace_record_result.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+// TraceRecordResult
+TraceRecordResult::TraceRecordResult(TraceType trace_type)
+    : trace_type_(trace_type) {}
+
+TraceRecordResult::~TraceRecordResult() {}
+
+TraceType TraceRecordResult::GetTraceType() const { return trace_type_; }
+
+// TraceExecutionResult
+TraceExecutionResult::TraceExecutionResult(uint64_t latency,
+                                           TraceType trace_type)
+    : TraceRecordResult(trace_type), latency_(latency) {}
+
+TraceExecutionResult::~TraceExecutionResult() {}
+
+uint64_t TraceExecutionResult::GetLatency() const { return latency_; }
+
+// StatusOnlyTraceExecutionResult
+StatusOnlyTraceExecutionResult::StatusOnlyTraceExecutionResult(
+    const Status& status, uint64_t latency, TraceType trace_type)
+    : TraceExecutionResult(latency, trace_type), status_(status) {}
+
+StatusOnlyTraceExecutionResult::~StatusOnlyTraceExecutionResult() {}
+
+Status StatusOnlyTraceExecutionResult::GetStatus() const { return status_; }
+
+Status StatusOnlyTraceExecutionResult::Accept(Handler* handler) {
+  assert(handler != nullptr);
+  return handler->Handle(*this);
+}
+
+// SingleValueTraceExecutionResult
+SingleValueTraceExecutionResult::SingleValueTraceExecutionResult(
+    const Status& status, const std::string& value, uint64_t latency,
```

Comment:
nit: pass by value then move:
```
SingleValueTraceExecutionResult::SingleValueTraceExecutionResult(
    Status status, std::string value, uint64_t latency,
    TraceType trace_type)
    : TraceExecutionResult(latency, trace_type),
      status_(std::move(status)),
      value_(std::move(value)) {}
```

Reply:
sorry, I didn't see there's an rvalue overload for `value`. Then just change that for `status`:

```
SingleValueTraceExecutionResult::SingleValueTraceExecutionResult(
    Status  status, const std::string& value, uint64_t latency,
    TraceType trace_type)
    : TraceExecutionResult(latency, trace_type),
      status_(std::move(status)),
      value_(value) {}
```

Reply:
Done